models/offer_template.go

```diff
@@ -109,34 +111,67 @@ func ListOfferTemplates(db runner.Connection, gameID string, mr *MixedMetricsRep
 }
 
 // InsertOfferTemplate inserts a new offer template into DB
-func InsertOfferTemplate(db runner.Connection, ot *OfferTemplate, mr *MixedMetricsReporter) (*OfferTemplate, error) {
-	_, err := GetEnabledOfferTemplateByKeyAndGame(db, ot.Key, ot.GameID, mr)
+func InsertOfferTemplate(db runner.Connection, ot *OfferTemplate, onlyInsert bool, mr *MixedMetricsReporter) (*OfferTemplate, error) {
+	olderOt, err := GetEnabledOfferTemplateByKeyAndGame(db, ot.Key, ot.GameID, mr)
+
+	if ot.Metadata == nil {
+		ot.Metadata = dat.JSON([]byte(`{}`))
+	}
 
 	if err != nil {
 		notFoundErr := HandleNotFoundError("OfferTemplate", map[string]interface{}{"Key": ot.Key}, err)
 		if err == notFoundErr {
```

Comment:
Add some comment explaining that if notFoundErr is equal to err (the error was not changed by HandleNotFoundError) it means that is is NOT a NotFoundError